File: backend/src/main/kotlin/com/example/backend/models/Vente.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "vente")
class Vente {
  @Id
  @Column(name = "id", nullable = false)
  var id: Long? = null

  @Column(name = "prixTotal")
  var prixTotal: Double? = 0.0

  @Column(name = "prixPercu")
  var prixPercu: Double? = 0.0

  @Column(name = "dateVente")
  var dateVente: LocalDateTime? = null

  @Column(name = "dateEncaissement")
  var dateEncaissement: LocalDateTime? = null

  @Column(name = "commentaire")
  var commentaire: String? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "malade_id")
  var malade: Malade? = null

  @Column(name = "etat", length = 16)
  var etat: String? = null

  @Column(name = "reference", length = 16)
  var reference: String? = null

  @Column(name = "nouveau_info", length = 16)
  var nouveauInfo: String? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id")
  var user: User? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "prescripteur_id")
  var prescripteur: Prescripteur? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "employe_id")
  var employe: Employe? = null

  @Column(name = "reduction", length = 32)
  var reduction: String? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "caisse_id")
  var caisse: Caisse? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var  supprimer: Int? = 0

  companion object {
    const val VENTE_COMPTANT = "COMPTANT"
    const val VENTE_ASSURANCE = "ASSURANCE"
    const val VENTE_CREDIT = "CREDIT"

    const val VENTE_TYPE_PAIEMENT_ELECTRONIQUE = "ELECTRONIQUE"
    const val VENTE_TYPE_PAIEMENT_ESPECE = "ESPECE"
    const val VENTE_TYPE_PAIEMENT_TICKET = "TICKET"
    const val VENTE_TYPE_PAIEMENT_MIXTE = "MIXTE"
  }
}

File: backend_py/pharmaPython/app/models/vente.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.api.v1.db import Base

class Vente(Base):
  __tablename__ = "vente"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True)
  prix_total = Column(Float, default=0.0)
  prix_percu = Column(Float, default=0.0)
  date_vente = Column(DateTime)
  date_encaissement = Column(DateTime)
  commentaire = Column(String)
  malade_id = Column(Integer, ForeignKey("malade.id"), nullable=True)
  etat = Column(String(16))
  reference = Column(String(16))
  nouveau_info = Column(String(16))
  user_id = Column(Integer, ForeignKey("user.id"))
  prescripteur_id = Column(Integer, ForeignKey("prescripteur.id"), nullable=True)
  employe_id = Column(Integer, ForeignKey("employe.id"))
  reduction = Column(String(32))
  caisse_id = Column(Integer, ForeignKey("caisse.id"))
  supprimer = Column(Integer, default=0)

  # malade = relationship("Malade")
  user = relationship("User")
  caisse = relationship("Caisse")
  # malade = relationship("Malade")
  prescripteur = relationship("Prescripteur")
  employe = relationship("Employe")
  # caisse = relationship("Caisse")
